### PR TITLE
fix: navigating user back to the welcome screen in case of an invalid token [ROAD-642]

### PR DESCRIPTION
**Pull request by attriaayush**
When providing an invalid/revoked token the API responds with error code `401` which not being handled correctly navigates the user to error screen. This PR fixes that problem and navigates the user back to the welcome screen where user can generate a new token.

### CODE REVIEW

File: src/snyk/base/modules/snykLib.ts [hunk 1]
```diff
@@ -37,18 +36,20 @@ export default class SnykLib extends BaseSnykModule implements ISnykLib {
...
-      await this.user.identify(this.snykApiClient, this.analytics);
-
       if (workspacePaths.length) {
         this.logFullAnalysisIsTriggered(manual);
 
         void this.startOssAnalysis(manual, false);
         await this.startSnykCodeAnalysis(workspacePaths, manual, false); // mark void, handle errors inside of startSnykCodeAnalysis()
       }
     } catch (err) {
+      console.log(err);
```
Comment (michelkaporin) [inline src/snyk/base/modules/snykLib.ts:52]: You don't need that

File: src/snyk/base/modules/snykLib.ts [hunk 2]
```diff
@@ -37,11 +36,12 @@ export default class SnykLib extends BaseSnykModule implements ISnykLib {
 
       await this.contextService.setContext(SNYK_CONTEXT.FEATURES_SELECTED, true);
 
+      await this.user.identify(this.snykApiClient, this.analytics);
+      await this.contextService.setContext(SNYK_CONTEXT.LOGGEDIN, true);
```
Comment (michelkaporin) [inline src/snyk/base/modules/snykLib.ts:40]: Is there any need to move `await this.contextService.setContext(SNYK_CONTEXT.LOGGEDIN, true);` to this line?

**Review by michelkaporin — APPROVED** [review]
LGTM with a minor comment 🚀

File: src/snyk/base/modules/snykLib.ts (hunk 2 above)
Comment (attriaayush) [reply src/snyk/base/modules/snykLib.ts]: ah yes, I think this was a hangover from the previous implementation. We don't need to make any changes within `SnykLib` class.

**Pull request merged**